File: Python/Strings/is_unique.py
# Checks if a string has all unique characters
def is_unique(str):
	if str is None:
		return True
	# String is not unique if length exceeds number of ASCII characters
	if len(str) > 128:
		return False
	# Dictionary to check occurrence of each ASCII character in the string
	chars = {}
	for char in str:
		if char in chars:
			return False
		chars[char] = True
	return True

File: Java/String/isUnique.java
// Checks if a string has all unique characters
static boolean isUnique(String str) {
	if(str == null) {
		return true;
	}
	// String is not unique if length exceeds number of ASCII characters
	if(str.length() > 128) {
		return false;
	}
	// Array to check occurrence of each ASCII character in the string
	boolean[] chars = new boolean[128];
	for(int i = 0; i < str.length(); i++) {
		int val = str.charAt(i);
		if(chars[val]) {
			return false;
		}
		chars[va] = true;
	}
	return true;
}
